### PR TITLE
Doc: Create guardian.md

### PR DESCRIPTION
**Pull request by bruce-riley**

### CODE REVIEW

File: docs/guardian.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+For a list of the guardian public RPC endpoints see `PublicRPCEndpoints` in [sdk/mainnet_consts.go](../sdk/mainnet_consts.go)
+
+## Observation Lifecycle
+
+An observation transitions through the following steps:
+
+1. The watcher observes a message published by the Wormhole core contract.
+
+2. On certain chains that do not have instant finality, the watcher waits until the block in which the observation was
```
Comment (evan-gray) [inline docs/guardian.md:92]: "in which the message was published"

File: docs/guardian.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   published reaches the appropriate state, such as the block is finalized, safe, or the integrator requested instant
+   finality.
+3. Whenever the watcher decides the block is finalized, it posts it to the processor via a golang channel.
+
+4. The processor performs integrity checks on the message (via the governor and accountant). Either of these
+   components may cause the observation to be delayed (the governor up to twenty-four hours, the accountant until
+   a quorum of guardians report the observation).
+
+5. Once the observation clears the integrity checks, the processor signs the observation using the guardian key and
```
Comment (evan-gray) [inline docs/guardian.md:101]: "Once the message"

File: docs/guardian.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+6. The processor batches observations, waiting no more than a second before publishing a batch. The batches are posted
+   to the p2p package for publishing as a signed observation batch. The processor also allows for certain observations
+   (specifically Pyth messages) to be published without batching. These are published immediately with a batch size of one.
+
+7. The p2p package posts the signed observation batches to gossip.
+
+8. Other guardians receive the observation. Each guardian aggregates the observation until it reaches quorum and
+   it has made the observation itself.
+
+9. Once an observation reaches quorum, the processor generates a VAA, signs it with its guardian key, and posts it
```
Comment (evan-gray) [inline docs/guardian.md:113]: rm "signs it with its guardian key" as the processor does not separately sign the VAA, it merely aggregates the quorum of signatures to form the VAA.

**Review by evan-gray — CHANGES_REQUESTED** [review]
Overall good, just some nits. Would be good to solidify the language in the Observation Lifecycle section to use
- "message" to refer to the publication from core contract sent by the watcher to the processor
- "observation" to refer to the message hash + signature created in the processor from a message and gossiped
I think I covered the appropriate changes in the comments.

File: docs/guardian.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+### Accountant
+
+The accountant package is responsible for interfacing with the accountant contracts on Gateway to verify that a given
+transfer will not exceed the available notional value for a chain. The accountant package interfaces with two contracts.
+The accountant contract is used to monitor Token Bridge transfers and the NTT-accountant is used to monitor NTT transfers.
+
+### Query Support
+
+The query package is used to process Wormhole Queries (also known as CCQ) requests, forwarding them to the appropriate watcher
```
Comment (bingyuyap) [inline docs/guardian.md:69]: "(also known as CCQ, which stands for Cross Chain Queries)" ?

File: docs/guardian.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+aggregating those observations with those reported by other guardians until quorum is reached, and publishing signed VAAs.
+
+For details on how to set up and run a guardian node see [operations.md](operations.md).
+
+## Components
+
+### Watchers
+
+The watchers are responsible for monitoring the connected chains for Wormhole messages. They listen for messages published
+by the Wormhole core contract and post them to the processor. They also handle requests to re-observe messages that may have
```
Comment (bingyuyap) [inline docs/guardian.md:13]: Do we want to explain how reobservation is being done?

**Review by bingyuyap — APPROVED** [review]
Thanks for writing this!

**Review by evan-gray — DISMISSED** [review]

File: docs/guardian.md (hunk 5 above)
Comment (bruce-riley) [reply docs/guardian.md:13]: Added a section at the end.

**Review by bingyuyap — DISMISSED** [review]

File: docs/guardian.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+The p2p package also maintains a separate pair of topics used for receiving and sending Wormhole Queries messages.
+The guardian joins both of these topics, but it only subscribes to the request topic. This means a given guardian
+does not see the Queries responses from other guardians. The guardian also applies libp2p filters so that it only
+processes requests from a select set of peers.
+
+### Processor
+
+The processor takes messages observed from the watchers and observations gossipped by other guardians. It aggregates these
+until an observation is seen by a quorum of guardians, including itself. It then publishes the signed VAA. The processor
+stores signed VAAs in an on-disk badgerDB instance. The processor also interfaces with the governor and account packages
```
Comment (panoel) [inline docs/guardian.md:53]: "account" or "accountant"?

**Review by panoel — DISMISSED** [review]

**Review by panoel — APPROVED** [review]

**Review by bingyuyap — APPROVED** [review]
LGTM

**Review by evan-gray — APPROVED** [review]

**Pull request merged**